sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/RowConverter.scala

```diff
@@ -167,8 +167,10 @@ object CosmosRowConverter
 
     private def convertStructToSparkDataType(schema: StructType, objectNode: ObjectNode) : Seq[Any] =
         schema.fields.map {
+            case StructField(CosmosTableSchemaInferer.RAW_JSON_BODY_ATTRIBUTE_NAME, StringType, _, _) =>
+                objectNode.toString
```

Comment:
The opposite is missing - basically special-casing if the raw json and the required system properties for id/pk exist to use those on write

Reply:
Makes sense